spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/StubRunnerExecutor.java

```diff
@@ -248,11 +248,22 @@ private void sendMessage(Contract groovyDsl) {
 		List<YamlContract> yamlContracts = yamlContractConverter.convertTo(Collections.singleton(groovyDsl));
 		YamlContract contract = yamlContracts.get(0);
 		setMessageType(contract, ContractVerifierMessageMetadata.MessageType.OUTPUT);
-		// TODO: Json is harcoded here
-		this.messageVerifierSender.send(
-				JsonOutput
-						.toJson(BodyExtractor.extractClientValueFromBody(body == null ? null : body.getClientValue())),
-				headers == null ? null : headers.asStubSideMap(), outputMessage.getSentTo().getClientValue(), contract);
+
+		Object payload = null;
+		if (body.getClientValue() instanceof FromFileProperty fromFile) {
+			if (fromFile.isByte()) {
+				payload = fromFile.asBytes();
+			}
+			else {
+				payload = fromFile.asString();
```

Comment:
If that's a string we could try to actually return a JSON representation (it always worked like this so maybe we shouldn't change that)

Reply:
I noticed that the `FromFile` is already checking if the contents are json, XML, or just text. Do all those other types need to be wrapped in json? or can they just go as is when this is coming from a file?

Reply:
Yeah, look that we're using the `BodyExtractor` mechanism to convert the body to a map. We should then convert it to JSON for jsons. For Xml we should parse it out as a XML. BodyExtractor will return a map or a list or just the body depending on what the user put there in the contract. We need to then represent it in the format that will be correct